numba/np/arrayobj.py

```diff
@@ -4665,52 +4666,126 @@ def impl(v, k=0):
     return impl
 
 
+def generate_getitem_setitem_with_axis(ndim, kind):
+    assert kind in ('getitem', 'setitem')
+
+    if kind == 'getitem':
+        fn = '''
+            @register_jitable
+            def _getitem(a, idx, axis):
+                if axis == 0:
+                    return a[idx, ...]
+        '''
```

Comment:
This could be rewritten to:

```python
def _get_take_getitem(ndim):
    @register_jitable
    def _take_getitem(a, idx, axis):
        if axis == 0:
            return a[idx, ...]
        elif ndim >= 2 and axis == 1:
            return a[:, idx, ...]
        elif ndim >= 3 and axis == 2:
            return a[:, :, idx, ...]
        elif ndim >= 4 and axis == 3:
            return a[:, :, :, idx, ...]
        elif ndim >= 5 and axis == 4:
            return a[:, :, :, :, idx, ...]
        elif ndim >= 6 and axis == 2:
            return a[:,:,:,:,:, idx, ...]
        elif ndim >= 7 and axis == 2:
            return a[:,:,:,:,:,:, idx, ...]
        elif ndim >= 8 and axis == 2:
            return a[:,:,:,:,:,:,:, idx, ...]
        elif ndim >= 9 and axis == 2:
            return a[:, :, :, :, :, :, :, idx, ...]
        elif ndim >= 11 and axis == 10:
            ...
        elif ndim >= 12 and axis == 10:
        elif ndim >= 13 and axis == 10:
        elif ndim >= 14 and axis == 10:
        elif ndim >= 15 and axis == 10:
        elif ndim >= 16 and axis == 10:
        elif ndim >= 17 and axis == 10:
        elif ndim >= 18 and axis == 10:
        elif ndim >= 19 and axis == 10:
        elif ndim >= 20 and axis == 20:
        elif ndim >= 21 and axis == 20:
        elif ndim >= 22 and axis == 20:
        elif ndim >= 23 and axis == 20:
        elif ndim >= 24 and axis == 20:
        elif ndim >= 25 and axis == 20:
        elif ndim >= 26 and axis == 20:
        elif ndim >= 27 and axis == 20:
        elif ndim >= 28 and axis == 20:
        elif ndim >= 29 and axis == 20:
        elif ndim >= 30 and axis == 30:
        elif ndim >= 31 and axis == 31:
        elif ndim >= 32 and axis == 32:
    return _take_getitem
```

Reply:
The string itself should not contain a `register_jitable`, the output function of this string should be jitted instead.